feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -452,8 +465,15 @@ bool target_radio_config_set2(const struct schema_Wifi_Radio_Config *rconf,
 	strncpy(phy, target_map_ifname(ifname), sizeof(phy));
 
 	if (changed->channel && rconf->channel) {
-		blobmsg_add_u32(&b, "channel", rconf->channel);
-		rrm_radio_rebalance_channel(rconf);
+		/* If radio channel config has changed then rebalance channel,
+		 * if not, that means the channel state changed due to a 
+		 * radar detection and shall remain in the backup channel */
+		if (is_cloud_channel_change(rconf) == 0) {
+			if(rrm_radio_rebalance_channel(rconf) == 0) {
+				blobmsg_add_u32(&b, "channel", rconf->channel);
+				radio_fixup_set_primary_chan(rconf->if_name, rconf->channel);
```

Comment:
Why does radio_fixup_set_primary_chan() call depend on whether the the rrm_radio_rebalance_channel() worked?

I guess it gives us retries if the rebalance fails.  However, I think the call to radio_fixup_set_primary_chan() in the state update probably prevents this.

Reply:
We compare the fixup chan with the radio config channel , if they are different we do a rebalance_channel. So in case the rebalance_channel fails, the opensync would retry to set the channel and since the fixup chan and radio config chan are different even now , it proceeds to rebalance_channel. The radio state update only copies the channel value from wireless uci.